esphome/components/gt911/touchscreen/gt911_touchscreen.cpp

```diff
@@ -49,27 +47,22 @@ void GT911Touchscreen::setup() {
   ESP_LOGCONFIG(TAG, "GT911 Touchscreen setup complete");
 }
 
-void GT911Touchscreen::loop() {
+void GT911Touchscreen::update_touches() {
   i2c::ErrorCode err;
   touchscreen::TouchPoint tp;
   uint8_t touch_state = 0;
   uint8_t data[MAX_TOUCHES + 1][8];  // 8 bytes each for each point, plus extra space for the key byte
 
-  if (!this->store_.available)
-    return;
-  this->store_.available = false;
-
   err = this->write(GET_TOUCH_STATE, sizeof(GET_TOUCH_STATE), false);
   ERROR_CHECK(err);
   err = this->read(&touch_state, 1);
   ERROR_CHECK(err);
   this->write(CLEAR_TOUCH_STATE, sizeof(CLEAR_TOUCH_STATE));
 
-  if ((touch_state & 0x80) == 0)
+  if (touch_state  == 0)
```

Comment:
```suggestion
  if ((touch_state & 0x80) == 0)
```

Reply:
@Clyde, why would you want that change back, In both case (touch_state & 0x80) and (touch_state & 0x07) would mean a return without changes.
Maybe should change it into:
```suggestion
  if ((touch_state & 0x87) == 0)
```

Reply:
Bit 7 must be set for the data to be valid. The lower 3 bits encode the number of touches, which can range from 0-5. Your change would mean that an invalid byte of e.g. 0x01 would be incorrectly processed as a single touch. The code as I had it was based on reading the datasheet and actual testing. The check for the most significant bit is required, and should be reinstated.

Reply:
okay i understand it now :D thanks for explaining